app/Console/Commands/EsIndexDocuments.php

```diff
@@ -98,55 +100,59 @@ protected function finish(array $indices, array $oldIndices)
         }
     }
 
-    /**
-     * Indexes and returns the names of the indices.
-     *
-     * @return array names of the indices indexed to.
-     */
-    protected function index()
+    private function indexGroup($name)
     {
         $indices = [];
-        foreach ($this->groups as $name) {
-            $indices = array_merge($indices, $this->indexGroup($name));
-        }
+        $newIndices = [];
+        $types = collect(static::ALLOWED_TYPES[$name]);
 
-        return $indices;
-    }
+        $allSame = $types->every(function ($type) use ($types) {
+            return $type::esIndexName() === $types->first()::esIndexName();
+        });
 
-    private function indexGroup($name)
-    {
-        $indices = [];
-        $types = static::ALLOWED_TYPES[$name];
+        if (!$allSame) {
+            $this->error("All types in group {$name} must have the same index.");
+
+            return [];
+        }
+
+        $first = $types->first();
+        $last = $types->last();
+        $alias = $first::esIndexName();
+        $indexName = "{$first::esIndexName()}{$this->suffix}";
+        $pretext = $this->inplace ? 'In-place indexing' : 'Indexing';
 
-        foreach ($types as $i => $type) {
+        foreach ($types as $type) {
             $count = $type::esIndexingQuery()->count();
             $bar = $this->output->createProgressBar($count);
 
-            $indexName = "{$type::esIndexName()}{$this->suffix}";
-            $pretext = $this->inplace ? 'In-place indexing' : 'Indexing';
             $this->info("{$pretext} {$type} into {$indexName}");
 
-            if (!$this->inplace && $i === 0) {
+            if (!$this->inplace && $type === $first) {
                 // create new index if the first type for this index, otherwise
                 // index in place.
                 $type::esIndexIntoNew(static::BATCH_SIZE, $indexName, function ($progress) use ($bar) {
                     $bar->setProgress($progress);
                 });
             } else {
-                $type::esReindexAll(static::BATCH_SIZE, 0, [], function ($progress) use ($bar) {
+                $options = ['index' => $indexName];
+                $type::esReindexAll(static::BATCH_SIZE, 0, $options, function ($progress) use ($bar) {
                     $bar->setProgress($progress);
                 });
             }
 
-            if ($i === 0) {
-                $indices[] = $type::esIndexName();
+            $bar->finish();
+
+            if (!$this->inplace
```

Comment:
or just outside the loop? (and maybe revert back using `=== 0` check for the first one)